src/libtools/graph.cc

```diff
@@ -0,0 +1,129 @@
+/*
+ *            Copyright 2009-2018 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <algorithm>
+#include <string>
+#include <votca/tools/graph.h>
+
+using namespace std;
+
+namespace votca {
+namespace tools {
+
+class GraphNode;
+
+void Graph::updateIds_(Graph& g) {
```

Comment:
?? why this method updates `id_` of `this` object and do the update of external `g` object?

Reply:
Good point I have corrected this because it is confusing.